plugins/fileservices/fileservices.cpp

```diff
@@ -3140,3 +3140,30 @@ FILESERVICES_API void FILESERVICES_CALL fsClearExpireDays(ICodeContext * ctx, co
     else
         throw makeStringExceptionV(0, "ClearExpireDays: Could not find logical file %s", lfn.str());
 }
+
+static int noCommonDef = NotFound;
+FILESERVICES_API bool FILESERVICES_CALL fsGetNoCommonDefault()
+{
+    // This function returns the value of the DFUServerProcess' 'noCommon' property, if it is exists. 
+    // Othervise it returns with true, the recent default value of 'noCommon'
+#ifdef _CONTAINERIZED
+    return getComponentConfigSP()->getPropBool("@noCommon", true);
+#else
+    CriticalSection  noCommonDefCrit;
```

Comment:
This CS can't be a local stack member. It will not protect any other re-entrant call, since they'll all have their own CS they can lock, needs to be a global that any thread coming here locks to lock others out.

Reply:
Changed